File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.cs
using System;

class Program
{
    static void Main() {
        string[] s = Console.ReadLine().Split(' ');
        int n = int.Parse(s[0]),m = int.Parse(s[1]);
        bool[,] ice = new bool[205,205];
        for(int i=0;i<m;i++)
        {
            s = Console.ReadLine().Split(' ');
            int x = int.Parse(s[0]),y = int.Parse(s[1]);
            ice[x,y] = ice [y,x] = true;
        }
        int ans = 0;
        for(int i=1;i<=n;i++)
        {
            for(int j=i+1;j<=n;j++)
            {
                if(ice[i,j]) continue;
                for(int k=j+1;k<=n;k++)
                {
                    if(ice[i,k]||ice[j,k]) continue;
                    ans++;
                }
            }
        }
        Console.Write(ans);
    }
}

File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.py
n,m = map(int,input().split())
ice = [[False for _ in range(205)] for _ in range(205)]

for i in range(m):
    x,y = map(int,input().split())
    ice[x][y] = True;
    ice[y][x] = True;
ans = 0
for i in range(1,n+1):
    for j in range(i+1,n+1):
        if ice[i][j] : continue
        for k in range(j+1,n+1):
            if ice[i][k] or ice[j][k]: continue
            ans+=1
print(ans)